app/middleware/postcodeValidator.js

```diff
@@ -0,0 +1,42 @@
+const log = require('../lib/logger');
+const postcodeValidator = require('../lib/postcodeValidator');
+const renderer = require('./renderer');
+const PostcodesIOClient = require('postcodesio-client');
+
+const PostcodesIO = new PostcodesIOClient();
+
+function handleError(error, postcode, res, next) {
+  renderer.postcodeError(error, postcode, res, next);
+}
+
+function render(result, postcode, req, res, next) {
+  if (!result) {
+    renderer.invalidPostcodePage(postcode, req, res);
+  } else {
+    next();
+  }
+}
+
+function validatePostcode(req, res, next) {
+  const postcode = res.locals.postcode;
+
+  if (postcodeValidator.isOutcode(postcode)) {
+    log.info('validate-outcode-skip');
+    next();
+  } else {
+    log.info('validate-postcode-start');
```

Comment:
I'd suggest removing this as timing of events should be done in a different manner.